plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchExecutor.groovy

```diff
@@ -94,8 +94,14 @@ class AzBatchExecutor extends Executor implements ExtensionPoint {
     protected void initBatchService() {
         config = AzConfig.getConfig(session)
         batchService = new AzBatchService(this)
-        // generate an account SAS token if missing
-        if( !config.storage().sasToken )
+
+        // Generate an account SAS token using activeDirectory configs
+        if (!config.storage().sasToken && config.activeDirectory().isConfigured()) {
+            config.storage().sasToken = AzHelper.generateContainerSas(workDir, config.storage().tokenDuration)
+        }
+
+        // Generate an account SAS token using storage account keys
+        if (!config.storage().sasToken && !config.activeDirectory().isConfigured())
             config.storage().sasToken = AzHelper.generateAccountSas(workDir, config.storage().tokenDuration)
```

Comment:
Consider using a ternary expression here so that you only check each condition once.

Reply:
Yup, good point, accommodated the feedback with https://github.com/nextflow-io/nextflow/pull/3132/commits/930cb3b05a01daf493bd390e83071b61f73a2a01